splice_machine/src/test/java/com/splicemachine/derby/utils/ShowCreateTableIT.java

```diff
@@ -0,0 +1,273 @@
+                ",\"B2\" INTEGER\n" +
+                ",\"A2\" INTEGER\n" +
+                ",CONSTRAINT T12_FK_1 FOREIGN KEY (A2,B2) REFERENCES T2(A2,B2) ON UPDATE NO ACTION ON DELETE NO ACTION) ", rs.getString(1));
+    }
+
+    @Test
+    public void testUnique() throws Exception
+    {
+        ResultSet rs = methodWatcher.executeQuery("call syscs_util.SHOW_CREATE_TABLE('SHOWCREATETABLEIT','T6')");
+        rs.next();
+        String ddl = rs.getString(1);
+        Pattern pattern = Pattern.compile("SQL\\d+");
+        Matcher m1 = pattern.matcher(ddl);
+        String csName = null;
+        while (m1.find())
+            csName = m1.group();
+        Assert.assertEquals("CREATE TABLE \"SHOWCREATETABLEIT\".\"T6\" (\n" +
+                "\"A6\" INTEGER\n" +
+                ",\"B6\" INTEGER\n" +
+                ",\"C6\" INTEGER\n" +
+                ",\"D6\" INTEGER\n" +
+                ", CONSTRAINT " + csName + " UNIQUE (D6), CONSTRAINT U_T6_1 UNIQUE (A6), CONSTRAINT U_T6_2 UNIQUE (B6,C6)) ", ddl);
+    }
+
+    @Test
+    public void testCheck() throws Exception
+    {
+        ResultSet rs = methodWatcher.executeQuery("call syscs_util.SHOW_CREATE_TABLE('SHOWCREATETABLEIT','T7')");
+        rs.next();
+        Assert.assertEquals("CREATE TABLE \"SHOWCREATETABLEIT\".\"T7\" (\n" +
+                "\"A7\" INTEGER\n" +
+                ",\"B7\" INTEGER\n" +
+                ",\"C7\" CHAR(1)\n" +
+                ", CONSTRAINT B7_CONSTRAINT CHECK (B7 > 100), CONSTRAINT C7_CONSTRAINT CHECK (C7 IN ('B', 'L', 'D', 'S','''','\"'))) ", rs.getString(1));
+    }
+
+    @Test
+    public void testDefaultValue() throws Exception
+    {
+        ResultSet rs = methodWatcher.executeQuery("call syscs_util.SHOW_CREATE_TABLE('SHOWCREATETABLEIT','T7')");
+        rs.next();
+        Assert.assertEquals("CREATE TABLE \"SHOWCREATETABLEIT\".\"T7\" (\n" +
+                "\"A7\" INTEGER\n" +
+                ",\"B7\" INTEGER\n" +
+                ",\"C7\" CHAR(1)\n" +
+                ", CONSTRAINT B7_CONSTRAINT CHECK (B7 > 100), CONSTRAINT C7_CONSTRAINT CHECK (C7 IN ('B', 'L', 'D', 'S','''','\"'))) ", rs.getString(1));
+    }
+
+    @Test
+    public void testNonExistSchema() throws Exception
+    {
+        try
+        {
+            methodWatcher.execute("call syscs_util.SHOW_CREATE_TABLE('nonexist','t1')");
+            Assert.fail("Expected to fail");
+        }
+        catch (SQLException e)
+        {
+            String sqlState = e.getSQLState();
+            Assert.assertEquals("42Y07",sqlState);
+        }
+    }
+
+    @Test
+    public void testNonExistTable() throws Exception
+    {
+        try
+        {
+            methodWatcher.execute("call syscs_util.SHOW_CREATE_TABLE('SHOWCREATETABLEIT','foo')");
+            Assert.fail("Expected to fail");
+        }
+        catch (SQLException e)
+        {
+            String sqlState = e.getSQLState();
+            Assert.assertEquals("42X05",sqlState);
+        }
+    }
+
+}
```

Comment:
This looks like it will be a very useful feature.  Here are some general comments:

1. Could we perhaps print a semi-colon ';' at the end of the CREATE TABLE output, so the output of SHOW CREATE TABLE commands could be run from another sqlshell script without having to manually add them?

2.  The output of the primary key columns does not always follow the same order as the original CREATE TABLE, which means the output doesn't show the correct primary key.  Example:

create table t2 (a int, b int, primary key(b,a));
splice> show create table t2;
DDL
---
CREATE TABLE "SPLICE"."T2" (
"A" INTEGER NOT NULL
,"B" INTEGER NOT NULL
, CONSTRAINT SQL190213163554700 PRIMARY KEY(A,B)) <--- should be (B,A)

3.  The output does not show any of the index DDL for indexes on the table, so we would still need to manually reconstruct the CREATE INDEX statements, so limits it's usefulness in setting up tests based on customers' tables.  Is there a possibility of adding the CREATE INDEX statements to the output?  If this JIRA is going on too long, maybe open another JIRA for it.  SHOW CREATE VIEW would be another good one.

Reply:
Thanks Mark reviewing!
1. Add a semi-colon at the end of DDL
2. It is a bug to be addressed. Good test.
3. We could file new Jiras for INDEX and VIEW.